src/lib/i18n.ts

```diff
@@ -7,12 +7,22 @@ export enum Language {
   Pa = 'pa',
 }
 
+// TODO @Harjot1Singh confirm code below for potential pitfalls or optimizations
+// @bhajneet has concerns that there is a shorter/easier to read solution
+// and that perhaps if a translation key is missing, there would be an error or such
 export const createTranslations = (
-  translations: Record<Language, Record<string, string>>,
+  translation: Record<string, Record<Language, string>>,
   namespace: string,
 ) => {
   Object.values( Language ).forEach( ( languageCode ) => {
-    i18n.addResourceBundle( languageCode, namespace, translations[ languageCode ] )
+    const phrases = Object.create( null )
+
+    Object.entries( translation ).forEach( ( [ key, value ] ) => {
+      phrases[ key ] = value[ languageCode ]
+    } )
+
+    console.log( languageCode, phrases )
+    i18n.addResourceBundle( languageCode, namespace, phrases )
   } )
 }

```

Comment:
I was thinking that this function could return an object of phrases that would be the namespaced strings that you could use in the component btw. I'll add something to show you what I mean